### ISSUE TITLE
Fix feature collision avoidance

### ISSUE THREAD

**Issue by apendleton** [mapbox/carmen#847]
### Context
The coalesce process returns a record for each feature we might want to return as part of a result, consisting of a tuple `(relev, score, x, y, id)`, where the ID is the 20 least-significant bits of the actual original feature's ID. We then attempt to retrieve the feature using the `(x, y, id)` component, but sometimes more than one result comes back, because more than one feature on the same tile (the same `(x, y)`) has the same 20 least-significant bits in its ID.

In that case, we first try to disambiguate using the score, by recomputing the 3-bit score value for each feature. If they're *still* tied, since we also know which phrase produced each match, we look at all the labels on the feature (all the `/^carmen:text.*/` properties, break out their synonyms, and see if any of them match the phrase text we got back.

... except this seems to not actually work. #845 made it not crash (under some circumstances it was trying to generate an invalid regular expression), but it will still fail to find the feature we actually want lots (or maybe most) of the time, because it just looks for an occurrence of the cover text inside one of the feature labels. Problems:
* it's case-sensitive (both was before #845 and still is), and all of the phrase text will have been lowercased while in all likelihood, the actual feature text won't have been; the unit test creates a synthetic circumstance under which the case is mixed on both sides, but this won't occur in practice
* it doesn't account for token replacement, which will have been applied to the cover text but not the feature text
* it doesn't account for other transformations that might have occurred to the cover text but not the feature text, such as `###`-ifying of address numbers and, as of recently, rewriting of cover text for intersection labels.

Options:
* we could effectively re-run the indexing process on a given feature to regenerate its phrases, the same way we do to identify `matching_text` -- we might even be able to re-use the same code (or maybe even only run the operation once, and use its results for both purposes?). This process works most of the time, but not every time for reasons we haven't really dug into much.
* we could use a Levenshtein or other fuzzy comparison to see which text is closest to the cover text, with the expectation that this will still give us the right feature despite possible transforms
* both, maybe, if neither is reliable enough alone?

Bottom line, it probably shouldn't be possible for us to retrieve multiple features and then choose to use none of them, as seems to be the case now. We should always pick one, even if our confidence that it's the right one is tenuous.

Separately, whenever we next rework the grid format, we should consider moving to 32-bit IDs to make collisions less likely.

### Next Steps

TBD

@mapbox/search
